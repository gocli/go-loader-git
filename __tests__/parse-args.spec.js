const parseArgs = require('../lib/parse-args')

describe('Parse Arguments', () => {
  it('handles empty list without fail', () => {
    expect(parseArgs()).toMatchSnapshot()
  })

  it('handles all flags', () => {
    expect(parseArgs('c m d --depth 3 --checkout dev --git /bin/git --keep-git'.split(' '))).toMatchSnapshot()
  })

  it('handles alias for keep-git', () => {
    expect(parseArgs(['cmd', '-k'])).toMatchSnapshot()
  })

  it('handles no-install flag', () => {
    expect(parseArgs(['cmd', '--no-install'])).toMatchSnapshot()
  })
})
