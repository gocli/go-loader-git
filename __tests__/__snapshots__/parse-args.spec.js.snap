// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse Arguments handles alias for keep-git 1`] = `
Object {
  "_": Array [
    "cmd",
  ],
  "install": true,
  "k": true,
  "keep-git": true,
}
`;

exports[`Parse Arguments handles all flags 1`] = `
Object {
  "_": Array [
    "c",
    "m",
    "d",
  ],
  "checkout": "dev",
  "depth": 3,
  "git": "/bin/git",
  "install": true,
  "k": true,
  "keep-git": true,
}
`;

exports[`Parse Arguments handles empty list without fail 1`] = `
Object {
  "_": Array [],
  "install": true,
  "k": false,
  "keep-git": false,
}
`;

exports[`Parse Arguments handles no-install flag 1`] = `
Object {
  "_": Array [
    "cmd",
  ],
  "install": false,
  "k": false,
  "keep-git": false,
}
`;
